File: samples_generated_code/samples_2/phi4/baseline_code_py/code_row_95.py
import hashlib

def calculate_checksum_and_save(file_path, output_file):
    with open(file_path, 'rb') as f:
        data = f.read()
    
    if not data:
        checksum = 0
    else:
        checksum = int(hashlib.md5(data).hexdigest(), 16) % (2**32)
    
    bytes_read = len(data)
    
    with open(output_file, 'w') as out_f:
        out_f.write(f"Bytes read: {bytes_read}\n")
        out_f.write(f"Checksum: {checksum}")

# Example usage
# calculate_checksum_and_save('input.txt', 'output.txt')

File: samples_generated_code/samples_2/athene/baseline_code_java/code_row_95.java
import java.io.*;
import java.util.zip.CRC32;

public class ChecksumWriter {
    public static void main(String[] args) {
        if (args.length < 2) {
            System.out.println("Usage: java ChecksumWriter <input-file> <output-file>");
            return;
        }

        String inputFile = args[0];
        String outputFile = args[1];

        try (FileInputStream fis = new FileInputStream(inputFile);
             FileOutputStream fos = new FileOutputStream(outputFile)) {

            byte[] buffer = new byte[4096];
            int bytesRead;
            CRC32 crc32 = new CRC32();
            while ((bytesRead = fis.read(buffer)) != -1) {
                if (bytesRead > 0) {
                    crc32.update(buffer, 0, bytesRead);
                }
            }

            long checksum = crc32.getValue();
            fos.write((bytesRead + " ").getBytes());
            fos.write(Long.toString(checksum).getBytes());

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}